bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/devices/bridge/BoschHttpClient.java

```diff
@@ -0,0 +1,182 @@
+    private String systempassword;
+
+    private Gson gson = new Gson();
+
+    public BoschHttpClient(String ipAddress, String systempassword, SslContextFactory sslContextFactory) {
+        super(sslContextFactory);
+        this.ipAddress = ipAddress;
+        this.systempassword = systempassword;
+    }
+
+    private String getCertFromSslContextFactory() throws KeyStoreException, CertificateEncodingException {
+        Certificate cert = this.getSslContextFactory().getKeyStore().getCertificate(BoschSslUtil.getBoschSHCId());
+        return Base64.getEncoder().encodeToString(cert.getEncoded());
+    }
+
+    public boolean isAccessPossible() {
+        try {
+            String url = this.createSmartHomeUrl("devices");
+            Request request = this.createRequest(url, GET);
+            ContentResponse contentResponse = request.send();
+            String content = contentResponse.getContentAsString();
+            logger.debug("Access check response complete: {} - return code: {}", content, contentResponse.getStatus());
+            return true;
+        } catch (InterruptedException | TimeoutException | ExecutionException e) {
+            logger.debug("Access check response failed because of {}!", e.getMessage());
+            return false;
+        }
+    }
+
+    public boolean doPairing() {
+        logger.trace("Starting pairing openHAB Client with Bosch SmartHomeController!");
+        logger.trace("Please press the Bosch SHC button until LED starts blinking");
+
+        ContentResponse contentResponse;
+        try {
+            String publicCert = getCertFromSslContextFactory();
+            logger.trace("Pairing this Client '{}' with SHC {}", BoschSslUtil.getBoschSHCId(), ipAddress);
+
+            // JSON Rest content
+            Map<String, String> items = new HashMap<>();
+            items.put("@type", "client");
+            items.put("id", BoschSslUtil.getBoschSHCId()); // Client Id contains the unique OpenHab instance Id
+            items.put("name", "oss_OpenHAB_Binding"); // Client name according to
+                                                      // https://github.com/BoschSmartHome/bosch-shc-api-docs#terms-and-conditions
+            items.put("primaryRole", "ROLE_RESTRICTED_CLIENT");
+            items.put("certificate", "-----BEGIN CERTIFICATE-----\r" + publicCert + "\r-----END CERTIFICATE-----");
+
+            String url = this.createPairingUrl();
+            Request request = this.createRequest(url, HttpMethod.POST, items).header("Systempassword",
+                    Base64.getEncoder().encodeToString(systempassword.getBytes(StandardCharsets.UTF_8)));
+
+            contentResponse = request.send();
+
+            logger.trace("Pairing response complete: {} - return code: {}", contentResponse.getContentAsString(),
+                    contentResponse.getStatus());
+            if (201 == contentResponse.getStatus()) {
+                logger.info("Pairing successful.");
```

Comment:
Please avoid using info logging, change this to debug instead.
```suggestion
                logger.debug("Pairing successful.");
```